src/Krizalys/Onedrive/Client.php

```diff
@@ -359,17 +359,59 @@ public function obtainAccessToken($clientSecret, $code)
      * Renews the access token from OAuth. This token is valid for one hour.
      *
      * @param string $clientSecret The client secret.
-     * @param string $redirectUri  The redirect URI.
      */
-    /*public function renewAccessToken($clientSecret, $redirectUri)
+    public function renewAccessToken($clientSecret)
     {
-        $url = self::TOKEN_URL
-            . '?client_id=' . $this->_clientId
-            . '&redirect_uri=' . (string) $redirectUri
-            . '&client_secret=' . (string) $clientSecret
-            . '&grant_type=' . 'refresh_token'
-            . '&code=' . (string) $code;
-    }*/
+        if (null === $this->_clientId) {
+            throw new \Exception('The client ID must be set to call renewAccessToken()');
+        }
+
+        if (null === $this->_state->token->refresh_token) {
+            throw new \Exception('The refresh token not set or not permission in \'wl.offline_access\' for renew token');
```

Comment:
Beware the grammar.

=> `'The refresh token is not set or no permission for \'wl.offline_access\' was given to renew the token'`